### PR TITLE
style: optimize ColorPicker code

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [x] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | ref: #42326 |
| 🇨🇳 Chinese | 无需纳入 ChangeLog |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at e11e07a</samp>

This pull request refactors and improves the code quality, readability, type safety, and debuggability of the color picker component and its subcomponents. It simplifies some syntax, reorders and annotates imports, adds type annotations, extracts constants, and renames some variables and props. It affects the files `components/color-picker/ColorPickerPanel.tsx`, `components/color-picker/components/ColorInput.tsx`, `components/color-picker/components/ColorPresets.tsx`, `components/color-picker/components/ColorTrigger.tsx`, `components/color-picker/hooks/useColorState.ts`, `components/color-picker/components/ColorAlphaInput.tsx`, `components/color-picker/components/ColorClear.tsx`, `components/color-picker/components/ColorHsbInput.tsx`, `components/color-picker/components/ColorRgbInput.tsx`, and `components/color-picker/interface.ts`.

### <samp>🤖 Generated by Copilot at e11e07a</samp>

*  Simplify and improve the `ColorPresets` component by using shorthand syntax for `onChange` ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-3abf1a34a643a7087dfbaf3039447778574de2b61e627f4c1218297fb961c70fL40-R43)), removing unnecessary `?.` operator ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-8d2a305db0c4945017603df77e97b2bf27dbde09aef2a8e2c2f7d10e67a3a822L60-R60)), adding type annotation for `activeKeys` ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-8d2a305db0c4945017603df77e97b2bf27dbde09aef2a8e2c2f7d10e67a3a822L42-R42)), renaming `activeKey` prop to `activeKeys` ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-8d2a305db0c4945017603df77e97b2bf27dbde09aef2a8e2c2f7d10e67a3a822L53-R53)), and reordering imports ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-8d2a305db0c4945017603df77e97b2bf27dbde09aef2a8e2c2f7d10e67a3a822L1-R5))
* Add `displayName` property to `ColorPickerPanel` component for better debugging and testing ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-3abf1a34a643a7087dfbaf3039447778574de2b61e627f4c1218297fb961c70fR57-R61))
* Refactor and optimize the `ColorInput` component by defining a constant `selectOptions` for the format select component ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-92ce74ec7f4cf4bb67a1fe8516714a57233367553933dd62c59ae8806fe43146L19-R25), [link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-92ce74ec7f4cf4bb67a1fe8516714a57233367553933dd62c59ae8806fe43146L55-R69)), replacing the `steppersRender` function with a `steppersNode` memoized variable ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-92ce74ec7f4cf4bb67a1fe8516714a57233367553933dd62c59ae8806fe43146L32-R49)), and reordering imports ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-92ce74ec7f4cf4bb67a1fe8516714a57233367553933dd62c59ae8806fe43146L1-R7))
* Refactor and optimize the `ColorTrigger` component by adding a type annotation for the `containerNode` memoized variable ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-a6ee2d10d78910c13966101dad864556005e359953d2cc1059773b0a8e7f57c5L23-R30)), renaming the `containerRender` variable to `containerNode` ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-a6ee2d10d78910c13966101dad864556005e359953d2cc1059773b0a8e7f57c5L42-R46)), and reordering imports ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-a6ee2d10d78910c13966101dad864556005e359953d2cc1059773b0a8e7f57c5L1-R4))
* Improve the `useColorState` hook by adding the `readonly` modifier to the return type ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-9576ead765b69ad9edf8f148171300a9ac514ef80167a23b0ecd7e0704439647L11-R14)), changing the initial state value to be of type `Color` ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-9576ead765b69ad9edf8f148171300a9ac514ef80167a23b0ecd7e0704439647L11-R14)), adding the `as const` assertion to the return value ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-9576ead765b69ad9edf8f148171300a9ac514ef80167a23b0ecd7e0704439647L35-R32)), and simplifying the `hasValue` function parameter ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-9576ead765b69ad9edf8f148171300a9ac514ef80167a23b0ecd7e0704439647L5-R5))
* Change the `PresetsItem` interface to use an array type instead of a generic type for the `colors` property ([link](https://github.com/ant-design/ant-design/pull/42329/files?diff=unified&w=0#diff-83dba3026e7dc41ad64d3a2f4d484eeb085f0c3798dc2316fbb90d23520fca6eL11-R14))

### CODE REVIEW

File: components/color-picker/components/ColorInput.tsx [hunk 1]
```diff
@@ -29,17 +34,19 @@ const ColorInput: FC<ColorInputProps> = (props) => {
...
       case ColorFormat.rgb:
-        return <ColorRgbInput value={value} onChange={onChange} prefixCls={prefixCls} />;
+        return <ColorRgbInput {...inputProps} />;
       case ColorFormat.hex:
+        return <ColorHexInput {...inputProps} />;
       default:
-        return <ColorHexInput value={value} onChange={onChange} prefixCls={prefixCls} />;
+        return <ColorHexInput {...inputProps} />;
     }
-  };
```
Comment (yoyo837) [inline components/color-picker/components/ColorInput.tsx:42]: 上面有个警告

Comment (li-jia-nan) [reply components/color-picker/components/ColorInput.tsx:42]: 是说覆盖率掉了那个吗？

Comment (yoyo837) [reply components/color-picker/components/ColorInput.tsx:42]: 是说没被用例覆盖到，应该需要修改用例或者新增用例

Comment (li-jia-nan) [reply components/color-picker/components/ColorInput.tsx:42]: 我是为了格式整齐，给 switch case 里面加了一行 return，多了这行导致覆盖率掉了，我又删掉了

Comment (RedJue) [reply components/color-picker/components/ColorInput.tsx:42]: > 我是为了格式整齐，给 switch case 里面加了一行 return，多了这行导致覆盖率掉了，我又删掉了

不加 return是为了让 hex 和 default 走同一个 case ，避免 eslint 的错误，同时也不掉测试覆盖率。

Comment (li-jia-nan) [reply components/color-picker/components/ColorInput.tsx:42]: > > 我是为了格式整齐，给 switch case 里面加了一行 return，多了这行导致覆盖率掉了，我又删掉了
> 
> 不加 return是为了让 hex 和 default 走同一个 case ，避免 eslint 的错误，同时也不掉测试覆盖率。

eslint 有什么报错我倒是没看到，那个 return 我删掉了，恢复了之前的格式

File: components/color-picker/ColorPickerPanel.tsx [hunk 2]
```diff
@@ -31,22 +31,11 @@ const ColorPickerPanel: FC<ColorPickerPanelProps> = (props) => {
...
       )}
       {panel}
-      <ColorInput
-        value={color}
-        onChange={(value) => onChange?.(value)}
-        prefixCls={prefixCls}
-        {...injectProps}
-      />
-
+      <ColorInput value={color} onChange={onChange} prefixCls={prefixCls} {...injectProps} />
```
Comment (li-jia-nan) [inline components/color-picker/ColorPickerPanel.tsx:34]: `value => onChange(value)` 可以简写成 `onChange`

File: components/color-picker/ColorPickerPanel.tsx [hunk 3]
```diff
@@ -31,22 +31,11 @@ const ColorPickerPanel: FC<ColorPickerPanelProps> = (props) => {
...
       {Array.isArray(presets) && (
         <>
           <Divider className={`${colorPickerPanelPrefixCls}-divider`} />
-          <ColorPresets
-            value={color}
-            presets={presets}
-            onChange={(value) => onChange?.(value)}
-            prefixCls={prefixCls}
-          />
+          <ColorPresets value={color} presets={presets} prefixCls={prefixCls} onChange={onChange} />
```
Comment (li-jia-nan) [inline components/color-picker/ColorPickerPanel.tsx:38]: 同上

File: components/color-picker/components/ColorInput.tsx [hunk 4]
```diff
@@ -29,17 +34,18 @@ const ColorInput: FC<ColorInputProps> = (props) => {
     setColorFormat(newFormat);
   };
 
-  const steppersRender = () => {
+  const steppersNode = useMemo<React.ReactNode>(() => {
```
Comment (li-jia-nan) [inline components/color-picker/components/ColorInput.tsx:37]: 这里用 useMemo 缓存一下，同时重命名了一下，`xxxxRender` 一般作为函数名，不适合做变量

File: components/color-picker/components/ColorInput.tsx [hunk 5]
```diff
@@ -52,24 +58,12 @@ const ColorInput: FC<ColorInputProps> = (props) => {
...
-          {
-            label: ColorFormat.hsb.toLocaleUpperCase(),
-            value: ColorFormat.hsb,
-          },
-          {
-            label: ColorFormat.rgb.toLocaleUpperCase(),
-            value: ColorFormat.rgb,
-          },
-        ]}
+        options={selectOptions}
```
Comment (li-jia-nan) [inline components/color-picker/components/ColorInput.tsx:61]: 这里是一个固定值，可以单独拿到组件外面去

File: components/color-picker/components/ColorInput.tsx (hunk 1 above)
Comment (RedJue) [reply components/color-picker/components/ColorInput.tsx:42]: > > > 我是为了格式整齐，给 switch case 里面加了一行 return，多了这行导致覆盖率掉了，我又删掉了
> 
> > 
> 
> > 不加 return是为了让 hex 和 default 走同一个 case ，避免 eslint 的错误，同时也不掉测试覆盖率。
> 
> 
> 
> eslint 有什么报错我倒是没看到，那个 return 我删掉了，恢复了之前的格式

之前为了不掉测试覆盖率，所以没加 default ，导致 eslint 报错，后面做了下兼容处理。

File: components/color-picker/components/ColorPresets.tsx [hunk 6]
```diff
@@ -39,7 +39,7 @@ const ColorPresets: FC<ColorPresetsProps> = ({ prefixCls, presets, value: color,
   });
   const colorPresetsPrefixCls = `${prefixCls}-presets`;
 
-  const activeKey = useMemo(
+  const activeKeys = useMemo<string[]>(
```
Comment (li-jia-nan) [inline components/color-picker/components/ColorPresets.tsx:42]: 重命名了一下，因为是数组，`activeKeys` 表示复数

File: components/color-picker/components/ColorTrigger.tsx [hunk 7]
```diff
@@ -20,14 +20,14 @@ const ColorTrigger = forwardRef<HTMLDivElement, colorTriggerProps>((props, ref)
   const { color, prefixCls, open, clearColor, disabled, className, ...rest } = props;
   const colorTriggerPrefixCls = `${prefixCls}-trigger`;
 
-  const containerRender = useMemo(
+  const containerNode = useMemo<React.ReactNode>(
```
Comment (li-jia-nan) [inline components/color-picker/components/ColorTrigger.tsx:23]: 同上，重命名了一下，`xxxxRender` 一般作为函数名，不适合做变量

File: components/color-picker/components/ColorTrigger.tsx [hunk 8]
```diff
@@ -20,14 +20,14 @@ const ColorTrigger = forwardRef<HTMLDivElement, colorTriggerProps>((props, ref)
...
     () =>
       clearColor ? (
         <ColorClear prefixCls={prefixCls} />
       ) : (
-        <ColorBlock color={color.toRgbString()} prefixCls={prefixCls} />
+        <ColorBlock prefixCls={prefixCls} color={color.toRgbString()} />
       ),
-    [color, clearColor],
+    [color, clearColor, prefixCls],
```
Comment (li-jia-nan) [inline components/color-picker/components/ColorTrigger.tsx:30]: 这里漏了个依赖项

File: components/color-picker/hooks/useColorState.ts [hunk 9]
```diff
@@ -2,19 +2,16 @@ import { useEffect, useState } from 'react';
 import type { Color } from '../color';
 import { generateColor } from '../util';
 
-function hasValue(value: Color | string | undefined) {
+function hasValue(value?: Color | string) {
```
Comment (li-jia-nan) [inline components/color-picker/hooks/useColorState.ts:5]: `undefined` 类型可以改成可选参数

File: components/color-picker/hooks/useColorState.ts [hunk 10]
```diff
@@ -2,19 +2,16 @@ import { useEffect, useState } from 'react';
...
 
 const useColorState = (
   defaultStateValue: Color | string,
-  option: {
-    defaultValue?: Color | string;
-    value?: Color | string;
-  },
-): [Color, React.Dispatch<React.SetStateAction<Color>>] => {
+  option: { defaultValue?: Color | string; value?: Color | string },
+): readonly [Color, React.Dispatch<React.SetStateAction<Color>>] => {
```
Comment (li-jia-nan) [inline components/color-picker/hooks/useColorState.ts:12]: 返回值改成 `readonly` 类型

File: components/color-picker/interface.ts [hunk 11]
```diff
@@ -8,7 +8,10 @@ export enum ColorFormat {
   hsb = 'hsb',
 }
 
-export type PresetsItem = { label: ReactNode; colors: Array<string | Color> };
+export interface PresetsItem {
```
Comment (li-jia-nan) [inline components/color-picker/interface.ts:11]: 这里不改也行，个人习惯，用 `interface` 描述对象，用 `type` 描述其它值或者函数

**Comment by RedJue** [comment]
这些修改我没意见👌

**Review by afc163 — APPROVED** [review]

**Pull request merged**